session/create_consumer.go

```diff
@@ -61,7 +63,13 @@ func (consumer *createConsumer) NewRequest() (requestPtr interface{}) {
 func (consumer *createConsumer) Consume(requestPtr interface{}) (response interface{}, err error) {
 	request := requestPtr.(*CreateRequest)
 
-	config, destroyCallback, err := consumer.configProvider(request.Config)
+	var pingerPort int
+	config, destroyCallback, err := consumer.configProvider(
+		request.Config,
+		func(port int) int {
```

Comment:
this confuses me, you're taking in a port, and returning the same port?

Reply:
here can be any number, but basically this `configProvider` spans as interface to service type configProviders and there we already return a value.

Reply:
made a comment in code.